manifest.go

```diff
@@ -59,6 +59,31 @@ func (f ManifestFile) parse() (manifest, error) {
 }
 
 func (m manifest) path(s string) (string, error) {
+	r, err := m.pathExact(s)
+	if err == nil || err == ErrEmpty {
+		return r, err
+	}
+	// If path references a runfile that lies under a directory that itself is a
+	// runfile, then only the directory is listed in the manifest. Look up all
+	// prefixes of path in the manifest.
+	prefixEnd := len(s)
+	for {
+		prefixEnd = strings.LastIndex(s[0:prefixEnd-1], "/")
+		if prefixEnd == -1 {
+			break
+		}
+		prefixMatch, err := m.pathExact(s[0:prefixEnd])
+		if err == ErrEmpty {
+			return "", ErrEmpty
+		}
+		if err == nil {
+			return prefixMatch + "/" + s[prefixEnd+1:], nil
+		}
+	}
+	return "", os.ErrNotExist
+}
+
+func (m manifest) pathExact(s string) (string, error) {
```

Comment:
This function is so trivial that it's probably better to inline it.
AIUI the empty case can't happen in line 75 since an empty RHS always indicates an empty file, never a directory.

Reply:
I think so too, changed.